src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -787,10 +787,18 @@ mongoc_server_description_handle_hello (mongoc_server_description_t *sd,
 
    EXIT;
 
-failure:
+typefailure:
    sd->type = MONGOC_SERVER_UNKNOWN;
    sd->round_trip_time_msec = MONGOC_RTT_UNSET;
+   bson_set_error (&sd->error,
+                   MONGOC_ERROR_STREAM,
+                   MONGOC_ERROR_STREAM_INVALID_TYPE,
+                   "invalid type sent to hello");
+   EXIT;
 
+authfailure:
+   sd->type = MONGOC_SERVER_UNKNOWN;
+   sd->round_trip_time_msec = MONGOC_RTT_UNSET;
```

Comment:
Suggestion to remove the duplication of setting `type` and `round_trip_time_msec`:

```suggestion
typefailure:
   bson_set_error (&sd->error,
                   MONGOC_ERROR_STREAM,
                   MONGOC_ERROR_STREAM_INVALID_TYPE,
                   "invalid type sent to hello");

authfailure:
   sd->type = MONGOC_SERVER_UNKNOWN;
   sd->round_trip_time_msec = MONGOC_RTT_UNSET;
```